source/geometry_model/initial_topography_model/prm_polygon.cc

```diff
@@ -0,0 +1,157 @@
+/*
+  Copyright (C) 2011 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/geometry_model/initial_topography_model/prm_polygon.h>
+#include <aspect/utilities.h>
+#include <boost/lexical_cast.hpp>
+
+namespace aspect
+{
+  namespace InitialTopographyModel
+  {
+    template <int dim>
+    double
+    PrmPolygon<dim>::
+    value (const Point<dim-1> &p) const
+    {
+      double value = 0;
+      Point<2> p1;
+
+      if (dim == 2)
+        p1 = Point<2>(p[0],0);
+      else
+        p1 = Point<2>(p[0],p[1]);
+
+      for (unsigned int i = 0; i < point_lists.size(); i++)
+        {
+          if (aspect::Utilities::polygon_contains_point<dim>(point_lists[i],p1))
+            {
+              value = topography_values[i];
+            }
+        }
```

Comment:
You are right it is inefficient. My point was mainly that the last should be chosen. It is more a conceptual thing for me, that you go through the list and 'paint over' in every polygon. I have made it more efficient now, in the sense that I turned around the for loop and put the return inside the if statement. 

Although it currently has my preference that the last one is the final one, but I do not feel very strongly about it. So if you really prefer it to be the other way around I am fine with it.